### PR TITLE
feat(configure): add initial config management support for connectivity features

### PR DESCRIPTION
**Pull request by DuckBoss**
This patch adds support for loading feature configurations from a drop-in config TOML file (/etc/rhc/config.toml.d/01-features.toml) with CLI flag override precedence.

The feature consolidation follows this precedence:
1. CLI flags (--enable/--disable) has highest priority
2. Drop-in config values are used as defaults if provided
3. Unspecified features or a lack of a drop-config file results in using built-in defaults

Note: Test cases were created with the help of AI tools to generate and evaluate edge cases and identify missing coverage.

Resolves: CCT-1804

### CODE REVIEW

File: cmd/rhc/main.go [hunk 1]
```diff
@@ -94,6 +95,17 @@ func beforeAction(c *cli.Context) error {
 
 	slog.SetLogLoggerLevel(conf.Config.LogLevel)
 
+	// Load features from drop-in config file and set them in conf.Config
+	// TODO: When drop-in configuration is fully supported, manually loading features
+	// may be unnecessary.
+	featuresConfig, err := features.GetFeaturesFromFile("/etc/rhc/config.toml.d/01-features.toml")
```
[the remark this answers is not in the document]
Comment (Archana-PandeyM) [reply cmd/rhc/main.go]: Should we  address this in another PR? I dont want to block the PR.
